client/landing/stepper/declarative-flow/internals/steps-repository/components/migration-survey/index.tsx

```diff
@@ -11,17 +13,22 @@ const linkByCountry = {
 
 type Countries = keyof typeof linkByCountry;
 
-type MigrationSurveyProps = Pick< SurveyProps, 'isOpen' > & {
-	countryCode: string;
-};
+type MigrationSurveyProps = Pick< SurveyProps, 'isOpen' >;
 
 const getLink = ( country: Countries ) => {
 	return linkByCountry[ country ];
 };
 
-const MigrationSurvey = ( { isOpen, countryCode }: MigrationSurveyProps ) => {
+const MigrationSurvey = ( { isOpen }: MigrationSurveyProps ) => {
+	const isEnLocale = useIsEnglishLocale();
+	const { data } = useGeoLocationQuery();
+	const countryCode = data?.country_short;
 	const surveyLink = getLink( countryCode as Countries );
 
+	if ( ! isEnLocale || ! surveyLink ) {
+		return null;
+	}
```

Comment:
The `! surveyLink` replaces the old `availableCountries` array, right? Unless the country is `IN` or `US`, the `surveyLink` will be `undefined`. I don't like the fact that the `getLink` types are muddy and kind of lying. The return type of `getLink` is `string`, but in practice it will be `string | undefined`. The `as Countries` type cast hides that. Can we fix that, preferrably by getting rid of the `Countries` type and replacing it with mere `string`?

Reply:
Made some revisions to the type handling of `getLink` in 88e4a02c9d745a66bcd2f48cf597339ca419ac2c . I found it a little tricky to maintain the `Countries` type since the incoming `countryCode` from `useGeoLocationQuery` is a `string` and would have to be cast explicitly. To @jsnajdr , I think it might just be better to work with them as strings, and we can still check if the country is defined in the links object.